VisualizationBase/src/items/ViewItem.cpp

```diff
@@ -158,6 +159,25 @@ QList<QPair<Item*, Item*>> ViewItem::arrowsForLayer(QString layer)
 	return arrows_[layer];
 }
 
+void ViewItem::determineChildren()
+{
+	Super::determineChildren();
+	//Update all the info nodes in the view
+	for (auto node : allNodes())
+	{
+		auto asViewItemNode = DCast<ViewItemNode>(node);
+		Q_ASSERT(asViewItemNode);
+		if (auto info = DCast<InfoNode>(asViewItemNode->reference()))
+		{
+			//Update only if we don't already do so by default
+			if (auto vis = findVisualizationOf(info))
+				//determineChildren is protected in Item, but not in DeclarativeItemBase
+				if (vis->needsUpdate() == Item::NoUpdate)
+					DCast<DeclarativeItemBase>(vis)->determineChildren();
```

Comment:
You should *not* call determineChildren() manually. Right now you're abusing the system by doing this and then checking for this special condition in that other method.

Why not:
* Here directly call `vis->node()->updateInfo(true)` for nodes that don't have a scheduled update.
* For the rest, let the standard mechanism do the update.

Or even better: You seem to need to update things on every cycle. I'm not quite clear on the distinction between doing `updateInfo(true)` and updateInfo(false)`. If we really need a distinction we could possibly implement this directly in the node itself and not here. What I would do here is simply to find all children that I want to update (in this case all info visualizations) and call setUpdateNeeded(StandardUpdate). Then let the method exit normally. The child items will be updated by the usual mechanism.

Please consider my last proposal and let me know what you think.